app/controllers/users/phone_confirmation_controller.rb

```diff
@@ -0,0 +1,97 @@
+module Users
+  class PhoneConfirmationController < ApplicationController
+    before_action :authenticate_user!
+    before_action :check_for_unconfirmed_mobile
+
+    def show
+      if FeatureManagement.prefill_otp_codes?
+        @code_value = confirmation_code
+      end
+
+      @unconfirmed_mobile = unconfirmed_mobile
+      @reenter_phone_number_path = if current_user.mobile.present?
+                                     edit_user_registration_path
+                                   else
+                                     users_otp_path
+                                   end
+    end
+
+    def new
+      send_new_confirmation_code
+      redirect_to phone_confirmation_path
+    end
+
+    def confirm
+      if params['code'] == confirmation_code
+        process_valid_code
+      else
+        process_invalid_code
+      end
+    end
+
+    def self.generate_confirmation_code
+      digits = Devise.direct_otp_length
+      random_base10(digits)
+    end
```

Comment:
This is probably fine for now, but we should eventually move the generation of the code to the 2FA gem, since it has very similar code. We could probably move much of this PR to the gem.